src/dimension_slice.c

```diff
@@ -1148,3 +1149,161 @@ ts_dimension_slice_get_chunkids_to_compress(int32 dimension_id, StrategyNumber s
 
 	return chunk_ids;
 }
+
+/* This function checks for overlap between the range we want to update
+ for the OSM chunk and the chunks currently in timescaledb (not managed by OSM)
+ */
+static bool
+ts_osm_chunk_range_overlaps(int32 osm_dimension_slice_id, int32 dimension_id, int64 range_start,
+							int64 range_end)
+{
+	bool res;
+	DimensionVec *vec = dimension_slice_collision_scan(dimension_id, range_start, range_end);
+	/* there is only one dimension slice for the OSM chunk. The OSM chunk may not
+	 * necessarily appear in the list of overlapping ranges because when first tiered,
+	 * it is given a range [max, infinity)
+	 */
+	if (vec->num_slices >= 2 ||
+		(vec->num_slices == 1 && vec->slices[0]->fd.id != osm_dimension_slice_id))
+		res = true;
+	else
+		res = false;
+	pfree(vec);
+	return res;
+}
+
+static ScanTupleResult
+dimension_slice_tuple_update(TupleInfo *ti, void *data)
+{
+	bool should_free;
+	HeapTuple tuple = ts_scanner_fetch_heap_tuple(ti, false, &should_free);
+	FormData_dimension_slice *fd = (FormData_dimension_slice *) data;
+
+	Datum values[Natts_dimension_slice] = { 0 };
+	bool isnull[Natts_dimension_slice] = { 0 };
+	bool doReplace[Natts_dimension_slice] = { 0 };
+
+	values[AttrNumberGetAttrOffset(Anum_dimension_slice_range_start)] =
+		Int64GetDatum(fd->range_start);
+	doReplace[AttrNumberGetAttrOffset(Anum_dimension_slice_range_start)] = true;
+
+	values[AttrNumberGetAttrOffset(Anum_dimension_slice_range_end)] = Int64GetDatum(fd->range_end);
+	doReplace[AttrNumberGetAttrOffset(Anum_dimension_slice_range_end)] = true;
+
+	HeapTuple new_tuple =
+		heap_modify_tuple(tuple, ts_scanner_get_tupledesc(ti), values, isnull, doReplace);
+
+	ts_catalog_update(ti->scanrel, new_tuple);
+
+	heap_freetuple(new_tuple);
+	if (should_free)
+		heap_freetuple(tuple);
+
+	return SCAN_DONE;
+}
+
+static int
+dimension_slice_update_by_id(int32 dimension_slice_id, FormData_dimension_slice *fd_slice)
+{
+	ScanKeyData scankey[1];
+
+	ScanKeyInit(&scankey[0],
+				Anum_dimension_slice_id_idx_id,
+				BTEqualStrategyNumber,
+				F_INT4EQ,
+				Int32GetDatum(dimension_slice_id));
+
+	return dimension_slice_scan_limit_internal(DIMENSION_SLICE_ID_IDX,
+											   scankey,
+											   1,
+											   dimension_slice_tuple_update,
+											   fd_slice,
+											   1,
+											   RowExclusiveLock,
+											   NULL,
+											   CurrentMemoryContext);
+}
+
+static int32
+ts_chunk_get_osm_slice_id(int32 chunk_id)
+{
+	Chunk *chunk = ts_chunk_get_by_id(chunk_id, true);
+	chunk->constraints =
+		ts_chunk_constraint_scan_by_chunk_id(chunk->fd.id, 1, CurrentMemoryContext);
+	Assert(chunk->constraints->num_constraints == 1);
```

Comment:
I'm getting an instance crash here.
```c
(gdb) p *chunk->constraints
$2 = {mctx = 0x55cf819a34f0, capacity = 5, num_constraints = 2, num_dimension_constraints = 1,
  constraints = 0x55cf819a4610}
```
I have a hypertable with a check constraint:
```sql
\set HYPERTABLE_NAME sample
CREATE TABLE :"HYPERTABLE_NAME"(
    f_int8 int8,
    f_int4 int4 DEFAULT 0 CHECK (f_int4 >= 0),
    f_int2 int2 GENERATED BY DEFAULT AS IDENTITY,
    f_date date GENERATED ALWAYS AS ('2022-01-01'::date + f_int4) STORED,
    f_timestamp timestamp
);
```

Reply:
oh thank you for pointing this out. What is the hypertable definition in this case?

Reply:
```
SELECT create_hypertable(:'HYPERTABLE_NAME', 'f_timestamp', chunk_time_interval=>'7 days'::interval);
```

Reply:
fixed and added a test case